astropy/utils/iers/iers.py

```diff
@@ -592,6 +592,49 @@ def pm_source(self, i):
         """Set PM source flag for entries in IERS table"""
         return np.ones_like(i) * FROM_IERS_B
 
+    @classmethod
+    def _create_a_columns(cls, table):
```

Comment:
> Thanks very much for this! Some initial comments. Note that I still haven't quite wrapped my head around the various things we actually want, and perhaps we should try to enumerate that. Is it roughly the following?
> 
>     1. `IERS_Auto` should make "best estimate from A and B" for all five columns, not just the three `UT1_UTC`, `PM_X` and `PM_Y` (included here).
> 
>     2. `IERS_Auto` without on-line connection should make a look-alike file from the IERS_B file (included here).
> 
>     3. It should be possible to have `IERS_B` be up to date (not included here).
> 
>     4. Any update on IERS_B should propagate to `IERS_Auto` (not included here).
> 
> 
> I think (1) is uncontroversial. (2) is less obvious to me - if we do it, I think _all_ columns should be populated, also `*_A` ones (or at least the column with the flag). I'd like to understand, though, whether it is useful to have those extra columns, or perhaps rather, whether it still would be useful if we did have (3) and possibly (4).

Hmm. Thinking about user desires:

- User wants a specific IERS A table, or IERS B table, or merged version, for reproducibility.
- User wants *some* kind of IERS data and has no connectivity.
- User wants an IERS A or B table that covers all the data in some particular data set.
- User wants the best available data that covers their data set.
- User wants IERS data to "just work" for reasonable times, past present and near future.
- User doesn't actually care much about accuracy but needs a stable set of IERS data for simulations possibly extending quite far into the future. (Observation planning, or generating par files for future observations of the triple system, for example.)
- User wants all astropy things that use IERS tables to use the same set of tables, whether autoupdated or not: having old the IERS data change partway through a computation because you requested an observatory position past the end of the cached table is liable to lead to some hair-tearing bugs that affect scientific results.

I'd like to see:
- A way to merge specific IERS A and IERS B tables to get a best-available combination.
- A way to get up-to-date-enough IERS A and IERS B tables (PINT needs IERS B specifically to agree with TEMPO2), preferably without user effort beyond perhaps making one request for all times up front.
- A way to have a "best-effort" IERS table - if the network is up, a merged version of auto-updated IERS A and B. Graceful degradation through the most recent A and B tables that exist in the cache to the plain B table distributed with astropy if the network is not available.
- A way to tell astropy to use specific table versions for either everything, or just everything inside this ``with`` block.

These four seem to cover all the use cases that sprang to my mind. Only the last, the context management, seems tricky to me, and that only if multithreading is to work right.

Reply:
This is a very nice summary, which deserves its own issue: #9227 (with to-do list added by me)

Reply:
Change made.